### PR TITLE
Switch CASDA to QueryWithLogin

### PR DESCRIPTION
**Pull request by jd-au**
This replaces the custom login in CASDA with use of QueryWithLogin.

### CODE REVIEW

File: astroquery/casda/tests/test_casda_remote.py
Comment (jd-au) [inline astroquery/casda/tests/test_casda_remote.py:55]: I've shown two approaches to credentials for the remote tests. However there is no keyring backend on the CI server, so I am thinking of removing this approach based around the keyring.

File: astroquery/casda/core.py
Comment (bsipocz) [inline astroquery/casda/core.py:327]: I think this was accidentally pulled in with one of  the merge commits, please rebase to get rid of those and any unrelated changes.

File: astroquery/casda/tests/test_casda_remote.py
Comment (bsipocz) [reply astroquery/casda/tests/test_casda_remote.py:55]: If there is any "test" user that we could use, I'm happy to set up github tokens that the CI will have access to. But that definitely seems to be something beyond this PR, so any approach that works for now sounds good on my end (e.g. you have real login credentials and say that these tests pass, I'm happy with it).

File: astroquery/casda/core.py
Comment (bsipocz) [inline astroquery/casda/core.py:57]: I'm a bit torn here as we're not at all consistent in the library as a whole, but having the password in the signature seems a bit of a counter-effective of security. I think I best like how the alma module is set up in this regard. What do you all think, am I just overly cautious? 

@keflavich  @andamian @pllim

File: astroquery/casda/core.py
Comment (jd-au) [reply astroquery/casda/core.py:327]: The commit was intentional - just cleaning up a minor aspect that was missed in the last PR

File: astroquery/casda/core.py
Comment (jd-au) [reply astroquery/casda/core.py:57]: I added the password param to allow testing of the login as there is no keyring backend in the CI environment. I couldn't see anyone else testing the login path. I agree the password param should not be used by other users, so perhaps either a "For testing use only" note in the method's docstring, or hiding it in a **kwargs would be better?

Comment (pllim) [reply astroquery/casda/core.py:57]: 👎 to explicit password argument in API. Might as well set up the backend to handle anonymous login without a password, or enable a more secure way (encrypted token) for the CI. Or just not test this at all in the CI and set up something private server-side to test this securely.

Comment (jd-au) [reply astroquery/casda/core.py]: Backend changes are out of scope for this piece of work unfortunately. I've removed the password parameter and the tests for the login code. For the other test code which requires login I have added a utility to simulate the login result.

File: astroquery/casda/tests/test_casda_remote.py
Comment (jd-au) [reply astroquery/casda/tests/test_casda_remote.py:55]: I've run these tests with real credentials and they pass. These remote tests now require the keyring so they can't be run on the github CI environment.

File: astroquery/casda/tests/test_casda.py
Comment (pllim) [inline astroquery/casda/tests/test_casda.py:142]:
```suggestion
    with pytest.raises(LoginError, match=r"If you do not pass a username to login\(\),") as excinfo:
        Casda.login()
```

File: astroquery/casda/core.py
Comment (pllim) [reply astroquery/casda/core.py]: Sounds good to me. Thanks!

Comment (bsipocz) [reply astroquery/casda/core.py]: Thank you. I opened https://github.com/astropy/astroquery/issues/2391 with the aim of tracking this password in API issue for the other affected modules, and finding an acceptable backend solution for testing (very much beyond the scope for this PR) has already have one opened one in #2367.

File: astroquery/casda/tests/test_casda.py
Comment (jd-au) [reply astroquery/casda/tests/test_casda.py]: I've made this change made all exception tests

File: astroquery/casda/core.py
Comment (jd-au) [reply astroquery/casda/core.py]: A consequence of removing the tests relying on the password parameter is that I can't achieve the code coverage target, so the CodeCov checks will fail unfortunately. The untested code occurs after the call to the keyring which would fail as there is no keyring backend in the GitHub CI environment.

Comment (bsipocz) [reply astroquery/casda/core.py]: yeah, no worries about the coverage test, those are not run with remote-data anyway, so always show a much smaller percentage as the reality.

I'll open a follow-up issue to get a backend installed in the VM, it shouldn't be too hard.

**Comment by bsipocz** [comment]
I'm sorry for merging the conflicting PR before this one, I missed noticing that it will cause issues. However, I'll go ahead and rebase this before merge to get a clear, linear history instead of the 3 merges.

**Comment by bsipocz** [comment]
(It looks like the email address you did these commits are not recognized by github, thus the weird avatar listing on them. It the git history it all look good, with you as author. I guess your avatar would also come up if you were to add that other email address to your GH account (you can have multiple ones associated)).

**Comment by jd-au** [comment]
> (It looks like the email address you did these commits are not recognized by github, thus the weird avatar listing on them. It the git history it all look good, with you as author. I guess your avatar would also come up if you were to add that other email address to your GH account (you can have multiple ones associated)).

Thanks for letting me know. I've corrected it locally now but will also update my GH account.

**Review by bsipocz — APPROVED** [review]
The remaining follow-up item requires to get keyring work on CI, so I would not hold this up any longer.

**Comment by bsipocz** [comment]
When running the tests with `remote-data=any`, the coverage is indeed much higher, 88%. 

I'll look into installing a keyring backend to CI, but that should not hold this one up.

**Comment by bsipocz** [comment]
Thank you @jd-au!

**Comment by jd-au** [comment]
Once the keyring backend is installed I'll be able to use the `cached_credentials` fixture from the remote tests to test the login code properly.

Thanks @bsipocz

**Pull request merged**
